## 2. Hypothesis Testing

In [8]:
import pandas as pd
from scipy.stats import shapiro, levene, ttest_ind, mannwhitneyu


In [10]:
df = pd.read_excel("fenerbahce_coaches_data.xlsx")


In [12]:
df["coach_type"] = df["is_foreign"].map({0: "Turkish", 1: "Foreign"})


In [15]:
turkish_ppg = df[df["coach_type"] == "Turkish"]["points_per_game"]
foreign_ppg = df[df["coach_type"] == "Foreign"]["points_per_game"]


In [17]:
print("Shapiro Turkish:", shapiro(turkish_ppg))
print("Shapiro Foreign:", shapiro(foreign_ppg))


Shapiro Turkish: ShapiroResult(statistic=np.float64(0.9660133240798583), pvalue=np.float64(0.8425988422392079))
Shapiro Foreign: ShapiroResult(statistic=np.float64(0.7397220432135939), pvalue=np.float64(0.002091698267197815))


In [18]:
levene_test = levene(turkish_ppg, foreign_ppg)
print("Levene Test:", levene_test)


Levene Test: LeveneResult(statistic=np.float64(0.0033367191522387946), pvalue=np.float64(0.9544352670694918))


In [19]:
if shapiro(turkish_ppg).pvalue > 0.05 and shapiro(foreign_ppg).pvalue > 0.05:
    equal_var = levene_test.pvalue > 0.05
    t_stat, p_value = ttest_ind(
        turkish_ppg,
        foreign_ppg,
        equal_var=equal_var
    )
    test_name = "Independent Samples t-test"
else:
    t_stat, p_value = mannwhitneyu(
        turkish_ppg,
        foreign_ppg,
        alternative="two-sided"
    )
    test_name = "Mann–Whitney U test"

print(test_name)
print("Statistic:", round(t_stat, 3))
print("p-value:", round(p_value, 4))


Mann–Whitney U test
Statistic: 100.0
p-value: 0.2419


# Hypothesis Testing Results
To examine whether there is a statistically significant difference in overall performance between Turkish and foreign coaches, an independent group comparison was conducted using points per game (PPG) as the performance metric.
Normality was assessed using the Shapiro–Wilk test. While the Turkish coaches’ PPG distribution did not significantly deviate from normality (p > 0.05), the foreign coaches’ distribution violated the normality assumption (p < 0.05). Therefore, a non-parametric Mann–Whitney U test was employed.
The Mann–Whitney U test indicated no statistically significant difference in points per game between Turkish and foreign coaches (U = 100.0, p = 0.2419).
As a result, the null hypothesis could not be rejected. This suggests that there is no statistically significant difference in overall performance between Turkish and foreign coaches based on points per game.
